src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1407,10 +1407,15 @@ public static string HexEscape(char character)
             {
                 throw new ArgumentOutOfRangeException(nameof(character));
             }
-            char[] chars = new char[3];
-            int pos = 0;
-            UriHelper.EscapeAsciiChar(character, chars, ref pos);
-            return new string(chars);
+
+            unsafe
```

Comment:
The code looks good but I would expect this kind of inlining optimization from a compiler rather from the developer.